src/dmd/dsymbolsem.d

```diff
@@ -555,6 +555,87 @@ private uint setMangleOverride(Dsymbol s, const(char)[] sym)
     return 0;
 }
 
+/*
+  Compiles the mixin content of CompileExp, CompileDeclaration
+  and CompileStatement.
+
+  Params:
+    CompileEntity = the type of mixin
+    sc            = the scope containing the mixin
+
+  Returs:
+    The resulting AST after the parsing phase
+*/
+auto compileIt(CompileEntity)(CompileEntity ce, Scope* sc)
+if (is(CompileEntity == CompileExp) ||
+    is(CompileEntity == CompileDeclaration) ||
+    is(CompileEntity == CompileStatement))
+{
+    static if (is(CompileEntity == CompileStatement))
```

Comment:
I think it would be better to unconditionally declare `errorStatements` and move the static if to return either null of an error statement inside that